### PR TITLE
Feature/update date picker

### PR DESCRIPTION
**Pull request by Zeljko-Predjeskovic**

### CODE REVIEW

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    public void TestMinimumDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.MinimumDate = new DateTime(1950, 1, 1);
+
+        Assert.Equal(new DateTime(1950, 1, 1), picker.MinimumDate);
+
+        
+        Assert.Throws<ArgumentException>(() => picker.MinimumDate = new DateTime(2200, 1, 1));
```
Comment (eszterhorvath) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:15]: this is actually 2 test cases ("fact"s) in one test

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+
+    [Fact]
+    public void TestNullableDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+        if (picker.Date == null)
```
Comment (eszterhorvath) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs]: why is this check necessary?

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        picker.PropertyChanged += (sender, e) =>
+        {
+            switch (e.PropertyName)
+            {
+                case "MaximumDate":
+                    maximumDateChanged = true;
+                    break;
+                case "Date":
+                    dateChanged = true;
+                    Assert.False(maximumDateChanged);
```
Comment (eszterhorvath) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:58]: why is this necessary?

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                    break;
+            }
+        };
+
+        var newDate = new DateTime(2000, 1, 1);
+        picker.MaximumDate = newDate;
+
+        Assert.True(maximumDateChanged);
+        Assert.True(dateChanged);
```
Comment (eszterhorvath) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:67]: also I don't get it, is you change the maximum date, why should the date also change?

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        Assert.True(minimumDateChanged);
+        Assert.True(dateChanged);
+
+        Assert.Equal(newDate, picker.MinimumDate);
+        Assert.Equal(newDate, picker.Date);
+        Assert.Equal(picker.MinimumDate, picker.Date);
+    }
+
+    [Fact]
+    public void TestDateClamping()
```
Comment (eszterhorvath) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:111]: image

**Review by eszterhorvath — APPROVED** [review]

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs (hunk 1 above)
Comment (gentledepp) [reply Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:15]: yeah.. please create separate tests here

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    [Fact]
+    public void TestMaximumDate()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.MaximumDate = new DateTime(2050, 1, 1);
+
+        Assert.Equal(new DateTime(2050, 1, 1), picker.MaximumDate);
+
+        Assert.Throws<ArgumentException>(() => picker.MaximumDate = new DateTime(1800, 1, 1));
```
Comment (gentledepp) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:35]: shouldn't you set the "picker.Date" to an invalid value causing an exception?
Or what are you actually testing here?

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+    [Theory, MemberData(nameof(DateTimesData))]
+    public void DatePickerSelectedEventArgs(DateTime initialDate, DateTime finalDate)
+    {
+        var datePicker = new Optiq.DatePicker.DatePicker();
+        datePicker.Date = initialDate;
+
+        Optiq.DatePicker.DatePicker pickerFromSender = null;
+        DateTime? oldDate = new DateTime();
```
Comment (gentledepp) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:159]: why don't you initialize those variables with `null`? `new DateTime()` looks very strange to me

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    public void SetNullValueDoesNotThrow()
+    {
+        var datePicker = new Optiq.DatePicker.DatePicker();
+        datePicker.SetValue(Optiq.DatePicker.DatePicker.DateProperty, null);
+        Assert.Null(datePicker.Date);
+    }
+
+    [Fact]
+    public void SetNullableDateTime()
+    {
```
Comment (gentledepp) [inline Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:187]: wouldn't it make more sense to set the `Date` property to `null`?

**Review by gentledepp — CHANGES_REQUESTED** [review]
please fix what we've commented!
Did you test this manually on Android and iOS too?

File: Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs (hunk 6 above)
Comment (Zeljko-Predjeskovic) [reply Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs:35]: testing that the maximum value isnt valid when its smaller than the minimum date

**Comment by Zeljko-Predjeskovic** [comment]
I copied the unit tests from maui to confirm that the datepicker works alike
